crypto/x509/x509_meth.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <time.h>
+#include <errno.h>
+
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/x509.h>
+#include <openssl/ossl_typ.h>
+#include "x509_lcl.h"
+
+X509_LOOKUP_METHOD *X509_LOOKUP_meth_new(const char *name)
+{
+    X509_LOOKUP_METHOD *method = OPENSSL_zalloc(sizeof(X509_LOOKUP_METHOD));
+
+    if (method != NULL) {
+        method->name = OPENSSL_strdup(name);
+        if (method->name == NULL) {
+            X509err(X509_F_X509_LOOKUP_METH_NEW, ERR_R_MALLOC_FAILURE);
+            goto err;
+        }
+    }
+
+    return method;
+
+err:
+    OPENSSL_free(method);
+    return NULL;
+}
+
+void X509_LOOKUP_meth_free(X509_LOOKUP_METHOD *method)
+{
+    if (method != NULL) {
+        OPENSSL_free(method->name);
+    }
+    OPENSSL_free(method);
+}
+
+int X509_LOOKUP_meth_set_new_item(
+    X509_LOOKUP_METHOD *method,
+    int (*new_item) (X509_LOOKUP *ctx))
+{
+    method->new_item = new_item;
+    return 1;
+}
+
+int (*X509_LOOKUP_meth_get_new_item(const X509_LOOKUP_METHOD* method))
+    (X509_LOOKUP *ctx)
+{
+    return method->new_item;
+}
+
+int X509_LOOKUP_meth_set_free(
+    X509_LOOKUP_METHOD *method,
+    void (*free) (X509_LOOKUP *ctx))
```

Comment:
The formatting here is not quite to our style. Please move the `X509_LOOKUP_METHOD *method` onto the end of
the previous line. If subsequent args won't fit on the same line, line them up with the first arg. So more like this:

````
int X509_LOOKUP_meth_set_free(X509_LOOKUP_METHOD *method,
                              void (*free) (X509_LOOKUP *ctx))
````

This same comment applies to numerous other function declarations in this file so I won't repeat for all of them.